src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader_MultiSegment.cs

```diff
@@ -105,16 +103,16 @@ private bool ReadFirstToken(ref BufferReader<byte> reader, byte first)
                 {
                     if (first == '-')
                     {
-                        //TODO: Is this a valid check?
-                        if (reader.End)
+                        // Couldn't find anything beyond '-', can't advance past it since we need '-' to be part of the value
+                        // TODO: This could silently allocate. Try to avoid it.
+                        if (reader.Peek(2).Length == 1)
```

Comment:
cc @JeremyKuhne, is there another approach here for me to validate the input without having to allocate (or advance/rewind)?